kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -129,16 +144,73 @@ class file_logger {
     std::unordered_map<uint16_t, std::unique_ptr<std::ofstream>> files;
 };
 
+/// Wrapper to the micctrl application, that controls the xeon phi coprocessor
+class mic_ctrl {
+public:
+    mic_ctrl(int adapter_)
+        :adapter(adapter_) {
+      }
+
+
+    void boot() {
+      std::stringstream ss;
+      ss << "sudo sh -c \"echo \\\"boot:elf:`pwd`/boot64.elf\\\" > /sys/class/mic/mic"
+          << adapter << "/state\"";
+      system(ss.str().c_str());
+    }
+
+    void reset_wait () {
+      std::stringstream ss;
+      ss << "sudo micctrl -r --wait mic" << adapter << " mic" << adapter;
+      system(ss.str().c_str());
+    }
+
+    void reset() {
+      std::stringstream ss;
+      ss << "sudo micctrl -r mic" << adapter;
+      system(ss.str().c_str());
+    }
+
+    void status() {
```

Comment:
We don't need this. The micctrl call was for debugging the state of the cards. Instead you can poll with read() on the state file's filedescriptor until it is "ready".